torch/distributed/distributed_c10d.py

```diff
@@ -4315,6 +4315,7 @@ def split_group(
     timeout: Optional[timedelta] = None,
     pg_options: Optional[Any] = None,
     group_desc: Optional[str] = None,
+    split_share: Optional[bool] = None,
```

Comment:
I would not add an option here because it is missed leading.

What matters is the `parent_pg` 's `split_share` setting. Setting it on the new communicators will only have an impact if you split again from those child communicators.
Also, by default, child communicators will inherit the config of the parent communicator.
See https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/api/types.html?highlight=splitshare

Reply:
Interesting, so we need an option for the root pg as well. But suppose we have this inheritance relationship: PG0->PG1->PG2, users want PG0, and PG1 to 'share' resources, PG1 does not want to further share resources with PG2, we still need this arg? just it needs to be clear that it is PG1's creator that 'stops' the further sharing

Reply:
So I would just handle setting the split_share setting via `pg_options`. So that it is consistent with the other nccl settings: `blocking`, `cga_cluster_size`, `min_ctas`, `max_ctas`, etc...

`pg_options` is already an arg of `init_process_group`, `split_group`, and the other APIs creating communicators...

Reply:
Do these configs, such as blocking, cga_cluster_size, min_ctas, max_ctas, can be configured per PG/comms, or it is the global config for all comms?

Reply:
They are configs per communicator.

Reply:
OK, that's good then, we can go this way